tests/snuba/api/endpoints/test_organization_events_mep.py

```diff
@@ -3220,32 +3226,33 @@ def test_is_metrics_extracted_data_is_included(self) -> None:
         response = self._make_on_demand_request(params)
         self._assert_on_demand_response(response)
 
-    @pytest.mark.skip(reason="Re-enable when user misery is supported again.")
-    def test_transaction_user_misery(self) -> None:
+    def test_on_demand_user_misery(self) -> None:
         user_misery_field = "user_misery(300)"
-        apdex_field = "apdex(300)"
-        params = {
-            "field": [user_misery_field, apdex_field],
-            "project": self.project.id,
-            "query": "",
-        }
-        specs = self._create_specs(params, groupbys=["transaction"])
-        for spec in specs:
-            self.store_on_demand_metric(1, spec=spec)
+        query = "transaction.duration:>=100"
+
+        spec = OnDemandMetricSpec(
+            field=user_misery_field,
+            query=query,
+            spec_type=MetricSpecType.DYNAMIC_QUERY,
+            # We only allow querying the function in the latest spec version,
+            # otherwise, the data returned by the endpoint would be 0.05
+            spec_version=OnDemandMetricSpecVersioning.spec_versions[-1],
+        )
+        tags = {"satisfaction": "miserable"}
+        self.store_on_demand_metric(1, spec=spec, additional_tags=tags, timestamp=self.min_ago)
+        self.store_on_demand_metric(2, spec=spec, timestamp=self.min_ago)
+
+        params = {"field": [user_misery_field], "project": self.project.id, "query": query}
+        self._create_specs(params)
+
+        # We expect it to be False because we're not using the extra feature flag
         response = self._make_on_demand_request(params)
+        self._assert_on_demand_response(response, expected_on_demand_query=False)
+
+        # Since we're using the extra feature flag we expect user_misery to be an on-demand metric
+        response = self._make_on_demand_request(params, {SPEC_VERSION_TWO_FLAG: True})
         self._assert_on_demand_response(response, expected_on_demand_query=True)
-        assert response.data == {
-            "data": [{user_misery_field: 0.0, apdex_field: 0.0}],
```

Comment:
The original test did not do a proper job of testing `user_misery` as it should have not been returning `0`.